compiler/packages/babel-plugin-react-compiler/src/Inference/InferMutationAliasingEffects.ts

```diff
@@ -2349,6 +2369,7 @@ function areArgumentsImmutableAndNonMutating(
     const kind = state.kind(place).kind;
     switch (kind) {
       case ValueKind.Primitive:
+      case ValueKind.ShallowMutable:
```

Comment:
I don’t think this is safe, a shallow mutable value can be mutated by the function